openedx/core/djangoapps/schedules/signals.py

```diff
@@ -78,6 +80,14 @@ def update_schedules_on_course_start_changed(sender, updated_course_overview, pr
     )
 
 
+@receiver(ENROLLMENT_TRACK_UPDATED)
+def reset_schedule_on_mode_change(sender, user, course_key, **kwargs):  # pylint: disable=unused-argument
+    """
+    When a CourseEnrollment's mode is changed, reset the user's schedule if self-paced.
+    """
+    reset_self_paced_schedule(user, course_key)
```

Comment:
Should we guard this for only verified mode or something? My current vibe is that regardless of why, if your mode changes, maybe you deserve a reset. But I don't know if that always holds?

Reply:
Hrm... Could be a problem if you can enroll as audit, upgrade to verified before your audit access disappears, and then refund to get back to audit with a reset schedule. Not sure if that's a loophole we care enough about, but if we do, maybe going to verified sets your schedule to the current date, and going to audit sets your schedule to start at the max of course start and enrollment date?

Reply:
I don't think there is an exploit there -- we are only resetting for self paced courses. And the user can do that at will right now, even in audit (right? we still show end dates and the reset button for audit?)

Reply:
I thought the plan was to only allow schedule resets for verified users, so as to not allow audit users to go past their original allowed audit time from FBE.

Reply:
Great call out, thanks Cale! I think I fixed that, please re-review.